src/web/components/SlotBodyEnd.tsx

```diff
@@ -20,10 +25,25 @@ const sendOphanInsertEvent = (): void => {
             name: 'remote_epic_test',
             variant: 'api',
         },
-        action: 'INSERT',
+        action,
     };
 
-    window.guardian.ophan.record({ componentEvent });
+    const key = JSON.stringify(componentEvent);
```

Comment:
A couple of random thoughts here:

Could the componentEvent ever include a function? If so, there's an error around stringifying/parsing functions which you'd hit

Could we get away with building a key as a string without `stringify`. It'd be less 'sure' but you'd get a mini performance win.

Reply:
> Could the componentEvent ever include a function?

Not as far as I can see, looking at the thrift definitions. I was thinking I needed something which uniquely identifies this Ophan call, which I think is the payload here. The performance question is a good one though, maybe I can use a subset of fields and get away without JSON encoding.